syntax = "proto3";


package wa.raft_kvserver; //所在的命名空间
option cc_generic_services = false;

// 日志实体
message GetRequest{
  bytes Key = 1 ;
  bytes ClientId = 2 ;
  int32 RequestId = 3;
}


message GetResponse  {
  //	下面几个参数和论文中相同
  bytes Err = 1;
  bytes Value = 2;
}


message SetRequest  {
  bytes  Key = 1;
  bytes  Value = 2 ;
  bytes  ClientId = 3;
  int32  RequestId = 4;
}

message SetResponse  {
  bytes Err = 1;
}

service kvServerRpc
{
  rpc Set (SetRequest) returns(SetResponse);
  rpc Get (GetRequest) returns (GetResponse);
}